components/camel-kudu/src/main/java/org/apache/camel/component/kudu/KuduConsumer.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.kudu;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.apache.camel.support.ScheduledPollConsumer;
+import org.apache.kudu.ColumnSchema;
+import org.apache.kudu.client.KuduClient;
+import org.apache.kudu.client.KuduException;
+import org.apache.kudu.client.KuduScanner;
+import org.apache.kudu.client.KuduTable;
+
+/**
+ * The Kudu consumer.
+ *
+ * @see org.apache.camel.support.ScheduledPollConsumer
+ * @see org.apache.camel.component.kudu.KuduEndpoint
+ */
+public class KuduConsumer extends ScheduledPollConsumer {
+    private final KuduEndpoint endpoint;
+
+    public KuduConsumer(KuduEndpoint endpoint, Processor processor) {
+        super(endpoint, processor);
+        this.endpoint = endpoint;
+    }
+
+    @Override
+    protected int poll() throws Exception {
+        Exchange exchange = endpoint.createExchange();
+
+        // create a message body
+        KuduScanner scanner = doScan(endpoint.getTableName());
+
+        exchange.getIn().setBody(scanner);
```

Comment:
IMHO is a bit weird to put the scanner directly in the message body (unless you have a different intention), hence do you think would it make sense to put `Iterator<RowResult>` as message body instead? Also, IMHO, I think it would be nice to include a type converter to `Map` as well (if possible of course).

Reply:
My first component and taking code from others, I am doing some things a bit blindly here :)

You are right, a good converter and conversion never hurts.

Reply:
No problems, I understand, it could a bit confusing at the beginning :). But, please don't hesitate to let us know if you need help or explaining anything, mailing list and gitter the best places ;)